lib/virtual-fs/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+use std::any::Any;
+use std::fmt;
+use std::io::{self, Read, Seek, Write};
+use std::path::Path;
+use thiserror::Error;
+use tracing::debug;
+
+pub mod host_fs;
+
+pub trait FileSystem: Send + Sync + 'static {
+    fn read_dir(&self, path: &Path) -> Result<std::fs::ReadDir, FsError>;
+    fn create_dir(&self, path: &Path) -> Result<(), FsError>;
+    fn remove_dir(&self, path: &Path) -> Result<(), FsError>;
+    fn rename(&self, from: &Path, to: &Path) -> Result<(), FsError>;
+
+    fn remove_file(&self, path: &Path) -> Result<(), FsError>;
+    fn new_open_options(&self) -> OpenOptions;
```

Comment:
What is this?

Reply:
Allows specifying how files can be opened, same as std::fs::OpenOptions